src/publishProject/publishProjectWebViewController.ts

```diff
@@ -95,6 +102,29 @@ export class PublishProjectWebViewController extends FormWebviewController<
         // Register reducers after initialization
         this.registerRpcHandlers();
 
+        // Listen for new connections
+        this.registerDisposable(
+            this._connectionManager.onConnectionsChanged(async () => {
```

Comment:
Please listen to onSuccessfulConnection, it gives you the connection info with the URI.